packages/puppeteer-core/src/api/locators/Locator.ts

```diff
@@ -113,33 +125,84 @@ export interface LocatorEventObject {
  * @public
  */
 export abstract class Locator<T> extends EventEmitter {
+  /**
+   * Creates a race between multiple locators but ensures that only a single one
+   * acts.
+   *
+   * @public
+   */
+  static race<Locators extends readonly unknown[] | []>(
+    locators: Locators
+  ): Locator<AwaitedLocator<Locators[number]>> {
+    return RaceLocator.create(locators);
+  }
+
   /**
    * Used for nominally typing {@link Locator}.
    */
   declare _?: T;
 
   /**
-   * Creates a race between multiple locators but ensures that only a single one
-   * acts.
+   * @internal
    */
-  static race<Locators extends Array<Locator<unknown>>>(
-    locators: Locators
-  ): Locator<UnionLocatorOf<Locators>> {
-    return new RaceLocator(
-      locators as Array<Locator<UnionLocatorOf<Locators>>>
-    );
-  }
+  protected visibility: VisibilityOption = null;
+  /**
+   * @internal
+   */
+  timeout = 30000;
+  #ensureElementIsInTheViewport = true;
+  #waitForEnabled = true;
+  #waitForStableBoundingBox = true;
 
   /**
-   * Creates an expectation that is evaluated against located values.
-   *
-   * If the expectations do not match, then the locator will retry.
-   *
    * @internal
    */
-  expect<S extends T>(predicate: Predicate<T, S>): Locator<S> {
-    return new ExpectedLocator(this, predicate);
-  }
+  protected operators = {
+    conditions: (
+      conditions: Array<Action<T, never>>,
+      signal?: AbortSignal
+    ): OperatorFunction<HandleFor<T>, HandleFor<T>> => {
+      return mergeMap((handle: HandleFor<T>) => {
+        return merge(
+          ...conditions.map(condition => {
+            return condition(handle, signal);
+          })
+        ).pipe(defaultIfEmpty(handle));
+      });
+    },
+    retryAndRaceWithSignalAndTimer: <T>(
+      signal?: AbortSignal
+    ): OperatorFunction<T, T> => {
+      const candidates = [];
+      if (signal) {
+        candidates.push(
+          fromEvent(signal, 'abort').pipe(
+            map(() => {
+              throw new Error('Locator aborted');
+            })
+          )
+        );
+      }
+      if (this.timeout > 0) {
+        candidates.push(
+          timer(this.timeout).pipe(
+            map(() => {
+              throw new TimeoutError(
+                `Timed out after waiting ${this.timeout}ms`
+              );
+            })
+          )
+        );
+      }
+      return pipe(
+        // For observables coming from promises, a delay is needed, otherwise
+        // RxJS will never yield in a permanent failure for a promise. (This is
+        // correct behavior).
+        retry({delay: 0}),
```

Comment:
should that be more than 0, e.g., `WAIT_FOR_FUNCTION_DELAY` we had before? currently, it looks like retry would be happening too often.

Reply:
Hmm, yes we can extend the delay further out. I'll add it.